numpyro/distributions/mixtures.py

```diff
@@ -320,11 +324,15 @@ def __init__(
         # TODO: It would be good to check that the support of all the component
         # distributions match, but for now we just check the type, since __eq__
         # isn't consistently implemented for all support types.
-        support_type = type(component_distributions[0].support)
-        if any(
-            type(d.support) is not support_type for d in component_distributions[1:]
-        ):
-            raise ValueError("All component distributions must have the same support.")
+        self._support = support
```

Comment:
In the property definition `def support(self):`, could you return `self._support` if it is not None? That is the main purpose of this `support` argument (to make sure that MCMC transform is defined correctly)